classes/patterns/class-homepage.php

```diff
@@ -0,0 +1,117 @@
+<?php
+/**
+ * HomePage pattern class.
+ *
+ * @package P4GBKS
+ * @since 0.1
+ */
+
+namespace P4GBKS\Patterns;
+
+/**
+ * This class is used for returning a home page template.
+ *
+ * @package P4GBKS\Patterns
+ */
+class HomePage extends Block_Pattern {
+
+	/**
+	 * @inheritDoc
+	 */
+	public static function get_name(): string {
+		return 'p4/home-page-pattern-layout';
+	}
+
+	/**
+	 * @inheritDoc
+	 *
+	 * @param array $params Optional array of parameters for the config.
+	 */
+	public static function get_config( $params = [] ): array {
+		return [
+			'title'      => __( 'Homepage', 'planet4-blocks-backend' ),
+			'blockTypes' => [ 'core/post-content' ],
+			'categories' => [ 'layouts' ],
+			'content'    => '
+			<!-- wp:group -->
+			<div class="wp-block-group">
+
+				<!-- wp:planet4-blocks/carousel-header -->
```

Comment:
TBH, I disagree to add the carousel as this way.
However, if we add it with somethig like <!-- wp:planet4-blocks/carousel-header { } --> you'll get a block error whenever you are editing this pattern.

It does happen something similar with the Covers.

Reply:
Do you think is possible to at least import this markup from somewhere else in the codebase? Because we would need to use Carousel Header in other page layouts too.

Reply:
Yes, I thought the same and also we need to use the Covers too.

Reply:
Maybe this CarouselHeader should be turned into a template too, similar to what we did with [Covers](https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/blob/master/classes/patterns/templates/class-covers.php)?

Reply:
Good point! I didn't add it since as far as I know this is the only pattern layout that uses this block. What do you think?

Reply:
I think it would still make sense to do, because it would make the pattern more readable and we might have other pattern layouts in the future that use the CarouselHeader too... But I'll leave it up to you, it's not a blocker at all 🙂

Reply:
Let's move it then 💪